src/client/metadataTransfer.ts

```diff
@@ -171,7 +165,7 @@ export abstract class MetadataTransfer<
       this.usernameOrConnection = await Connection.create({
         authInfo: await AuthInfo.create({ username: this.usernameOrConnection }),
       });
-      if (this.apiVersion) {
+      if (this.apiVersion && this.apiVersion !== this.usernameOrConnection.version) {
```

Comment:
Only set it and report that it's being overridden when the version has been set and is different.